pkg/controllers/staticroute/staticroute_controller.go

```diff
@@ -97,6 +98,10 @@ func (r *StaticRouteReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		if err := r.Service.CreateOrUpdateStaticRoute(req.Namespace, obj); err != nil {
 			updateFail(r, &ctx, obj, &err)
 			// TODO: if error is not retriable, not requeue
+			apierror, errortype := util.DumpAPIError(err)
```

Comment:
A nit suggestion, if apierror is not nil, could we write it to status, so we can debug from CR status more straight forward?